tensorflow_lite_support/ios/task/audio/core/audio_record/sources/TFLAudioRecord.m

```diff
@@ -0,0 +1,269 @@
+      AVAudioPacketCount inNumberOfPackets, AVAudioConverterInputStatus *_Nonnull outStatus) {
+    *outStatus = AVAudioConverterInputStatus_HaveData;
+    return pcmBuffer;
+  };
+
+  NSError *conversionError = nil;
+  AVAudioConverterOutputStatus converterStatus = [audioConverter convertToBuffer:outPCMBuffer
+                                                                           error:&conversionError
+                                                              withInputFromBlock:inputBlock];
+
+  switch (converterStatus) {
+    case AVAudioConverterOutputStatus_HaveData: {
+      return outPCMBuffer;
+    }
+    case AVAudioConverterOutputStatus_Error: {
+      NSString *errorDescription = conversionError.localizedDescription
+                                       ? conversionError.localizedDescription
+                                       : @"Some error occured while processing incoming audio "
+                                         @"frames.";
+      [TFLCommonUtils createCustomError:error
+                             withDomain:TFLAudioRecordErrorDomain
+                                   code:TFLAudioRecordErrorCodeProcessingError
+                            description:errorDescription];
+      break;
+    }
+    case AVAudioConverterOutputStatus_EndOfStream: {
+      [TFLCommonUtils createCustomError:error
+                             withDomain:TFLAudioRecordErrorDomain
+                                   code:TFLAudioRecordErrorCodeProcessingError
+                            description:@"Reached end of input audio stream."];
+      break;
+    }
+    case AVAudioConverterOutputStatus_InputRanDry: {
+      [TFLCommonUtils createCustomError:error
+                             withDomain:TFLAudioRecordErrorDomain
+                                   code:TFLAudioRecordErrorCodeProcessingError
+                            description:@"Not enough input is available to satisfy the request."];
+      break;
+    }
+  }
+  return nil;
+}
+
+- (BOOL)loadAudioPCMBuffer:(AVAudioPCMBuffer *)pcmBuffer error:(NSError **)error {
+  if (pcmBuffer.frameLength == 0) {
+    [TFLCommonUtils createCustomError:error
+                           withDomain:TFLAudioRecordErrorDomain
+                                 code:TFLAudioRecordErrorCodeInvalidArgumentError
+                          description:@"You may have to try with a different "
+                                      @"channel count or sample rate"];
+  } else if ((pcmBuffer.frameLength % self.audioFormat.channelCount) != 0) {
+    [TFLCommonUtils createCustomError:error
+                           withDomain:TFLAudioRecordErrorDomain
+                                 code:TFLAudioRecordErrorCodeInvalidArgumentError
+                          description:@"You have passed an unsupported number of channels."];
+  } else {
+    TFLFloatBuffer *floatBuffer = [[TFLFloatBuffer alloc] initWithData:pcmBuffer.floatChannelData[0]
```

Comment:
This looks like you're taking only the first channel from the pcmBuffer. In the case that AVAudioPCMBuffer has more than 1 channel, wouldn't the rest of the channel be filled with zeros?

Reply:
We are requesting the audio converter to return channels in an interleaved format. In this case both floatChannelData[0] and floatChannelData[1] point to the same 1d array with both channels in an interleaved format. You can check the [doc](https://developer.apple.com/documentation/avfaudio/avaudiopcmbuffer/1386212-floatchanneldata). I also tested this out.

Reply:
I see. Could you leave a comment in the doc to make it clearer? This behavior isn't obvious so we should leave a note for future maintainers of the library.